packages/flutter_tools/lib/src/devtools_launcher.dart

```diff
@@ -55,21 +55,30 @@ class DevtoolsServerLauncher extends DevtoolsLauncher {
         _dartExecutable,
         'devtools',
         '--no-launch-browser',
+        // TODO(kenz): add the `--print-dtd-uri` flag here once the Dart CLI
+        // supports it. See https://github.com/dart-lang/sdk/issues/55034.
+        // if (printDtdUri ?? false) '--print-dtd-uri',
         if (vmServiceUri != null) '--vm-uri=$vmServiceUri',
         ...?additionalArguments,
       ]);
       _processStartCompleter.complete();
-      final Completer<Uri> completer = Completer<Uri>();
+
+      final Completer<Uri> devToolsCompleter = Completer<Uri>();
       _devToolsProcess!.stdout
           .transform(utf8.decoder)
           .transform(const LineSplitter())
           .listen((String line) {
-            final Match? match = _serveDevToolsPattern.firstMatch(line);
-            if (match != null) {
-              final String url = match[1]!;
-              completer.complete(Uri.parse(url));
-            }
-         });
+        final Match? dtdMatch = _serveDtdPattern.firstMatch(line);
+        if (dtdMatch != null) {
+          final String uri = dtdMatch[1]!;
+          dtdUri = Uri.parse(uri);
```

Comment:
nit, we can return here and not bother checking for `_serveDevToolsPattern`, right?

Reply:
No these are independent of one another. The `_serveDevToolsPattern` should always be in the stdout. The `_serveDtdPattern` _might_ be there depending on whether the `--print-dtd-uri` flag was passed to the `dart devtools` command above.

Reply:
Would they be on the same line though?

Reply:
No these will be on separate lines.